core/src/main/java/io/parsingdata/metal/token/Def.java

```diff
@@ -44,20 +44,21 @@ public Def(final String name, final ValueExpression size, final Expression pred,
 
     @Override
     protected ParseResult parseImpl(final String scope, final Environment env, final Encoding enc) throws IOException {
-        final OptionalValue size = _size.eval(env, enc);
-        if (!size.isPresent()) {
+        final OptionalValueList size = _size.eval(env, enc);
+        if (size.size != 1) { throw new RuntimeException("Size must yield a single value."); }
+        if (!size.head.isPresent()) {
             return new ParseResult(false, env);
         }
-        // TODO: Handle value expression results as BigInteger (METAL-15)
-        final int dataSize = size.get().asNumeric().intValue();
+        // TODO: Handle value expression results as BigInteger
+        final int dataSize = size.head.get().asNumeric().intValue();
         if (dataSize < 0) {
             return new ParseResult(false, env);
         }
         final byte[] data = new byte[dataSize];
         if (env.input.read(env.offset, data) != data.length) {
             return new ParseResult(false, env);
         }
-        final Environment newEnv = new Environment(env.order.add(new ParseValue(scope, _name, this, env.offset, data, enc)), env.input, env.offset + size.get().asNumeric().intValue());
+        final Environment newEnv = new Environment(env.order.add(new ParseValue(scope, _name, this, env.offset, data, enc)), env.input, env.offset + size.head.get().asNumeric().intValue());
```

Comment:
Use `dataSize` here instead of calling `size.head.get().asNumeric().intValue()` again.